core/src/ldml/ldml_processor.cpp

```diff
@@ -519,14 +524,40 @@ ldml_event_state::ldml_event_state(
   this->modifier_state = m;
   this->is_key_down    = i;
   this->event_flags    = e;
+
+  actions.persist_options = new km_core_option_item[1];
+  actions.persist_options[0] = NULL_OPTIONS[0];
+
+  // initialize actions
+  clear();
 }
 
 void ldml_event_state::commit() {
-  state->actions().commit();
+  // TODO-LDML: is this necessary? what's the lifecycle of actions.output?
+  actions.output = new km_core_usv[text.length()+1];
+  memcpy(actions.output, text.c_str(), text.length()+1);
+  // current implementation copies actions.output and doesn't retain it
+  state->set_actions(actions);
+  // clean up
+  delete [] actions.output;
+  actions.output = nullptr;
```

Comment:
I'd move that into `clear()`

Reply:
Hmm, then I'm puzzled about the lifecycle

Reply:
Technically it's fine here, but it just seemed like `clear()` was a sensible place to flush out the struct entirely (same goes for the persist_options and where they are destroyed). But you own your actions_struct. So you can do what you want with it. set_actions will make a copy of what it needs -- ownership is not transferred.

Reply:
clear clears text() which is what i'm managing to hold the text string